docs/chapter_tree/avl_tree.md

```diff
@@ -0,0 +1,308 @@
+# AVL 树
+
+## AVL 树的起源
+
+在「二叉搜索树」章节中提到，如进行多次插入与删除操作后，二叉搜索树可能会退化为链表。此时所有操作的时间复杂度都会由 $O(\log n)$ 劣化至 $O(n)$ 。例如，删除结点 4 后，该二叉搜索树就会退化为链表。
+
+=== "删除前"
+    ![binary search tree1](avl_tree.assets/binary_search_tree1.png)
+=== "删除后"
+    ![binary search tree2](avl_tree.assets/binary_search_tree2.png)
+
+为了解决这一问题，G. M. Adelson-Velsky 和 E. M. Landis 在其 1962 年发表的论文 "An algorithm for the organization of info
+rmation" 中提出了「平衡二叉搜索树」，也以两位作者命名，常被称为「AVL 树」。
+
+## AVL 树的性质
+
+「AVL 树」既是「二叉搜索树」又是「平衡二叉树」，同时满足这两种二叉树的所有性质。「平衡二叉树」规定树中任意结点左右子树的高度差的绝对值不能超过 1 。本文定义：
+
+- 「平衡因子 Balance Factor」为 **左子树的高度减右子树的高度** 。
+- 空树的高度定义为 0 ，叶结点的高度定义为 1 。
```

Comment:
> The height of a node is the number of edges from the node to the deepest leaf.

Obviously, the height of a leaf node is 0.

I believe should not define a common definition to avoid unnecessary ambiguity.